src/ui/handler/scroll_zoom.js

```diff
@@ -356,6 +377,50 @@ class ScrollZoomHandler {
     reset() {
         this._active = false;
     }
+
+    /**
+     * Returns the scroll zoom blocker alert's HTML element.
+     *
+     * @example
+     * // Change the scroll zoom blocker element's font size
+     * map.scrollZoom.getScrollZoomBlockerElement();
+     * scrollZoomBlockerElem.style.fontSize = "25px";
+     * @returns {HTMLElement} Returns container element.
+     */
+
+    getScrollZoomBlockerElement() {
+        if (this._container) {
+            return this._container;
+        }
+    }
+
+    _addScrollZoomBlocker() {
+        if (!this._map) { return; }
+        if (!this._container) {
+            this._container = DOM.create('div', 'mapboxgl-scroll-zoom-blocker-control', this._map._container);
+            const alertMessage = this._getBlockerAlertMessage();
+            const textNode = window.document.createTextNode(alertMessage);
+            this._container.appendChild(textNode);
+        }
+    }
+
+    _getBlockerAlertMessage() {
+        return this._map._getUIString('ScrollZoomBlocker.Message');
+    }
+
+    _showBlockerAlert() {
+        this._container.classList.remove('mapboxgl-scroll-zoom-blocker-control-fade');
+        this._container.style.opacity = '1';
+        this._container.style.visibility = 'visible';
+    }
+
+    _fadeOutBlockerAlert() {
+        setTimeout(() => {
+            this._container.classList.add('mapboxgl-scroll-zoom-blocker-control-fade');
+            this._container.style.opacity = '0';
+        }, 2000);
```

Comment:
@mourner I had replaced the setTimeout with an event listener for transitionend event, but unfortunately it only worked as expected in Chrome, and failed to set off in firefox and safari. I still needed to add something to either listen to or wait for the css changes, otherwise it wouldn't have time to show the change. The transitionend event listener did not work as I needed it to because the way the scroll zoom blocker is set up in the wheel event, it is called very quickly multiple times.  In safari and firefox, if it the transition is the same as how it started, the transitionend event listener stops firing altogether which occurred in this scenario.... so I reverted back to using setTimeout, is this a deal breaker? And if so, is there something else I could use to give time for the css changes to appear in between adding and removing the classes? Thanks so much!

Reply:
@mourner addressed the changes aside from the setTimeout as mentioned in the previous comment. Let me know if this looks good to go or if there are changes to be made still. Thanks!